src/components/user/AddFile.tsx

```diff
@@ -0,0 +1,226 @@
+"use client";
+
+import React, { Dispatch, SetStateAction, useState } from "react";
+import FilterDropdown from "@components/FilterDropdown";
+import Tag, { TagProps, tagList } from "@components/Tag";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { createFile } from "@api/file/route.client";
+import { File as PrismaFile, Prisma } from "@prisma/client";
+import { Checkbox, button } from "@material-tailwind/react";
+import { setActive } from "@material-tailwind/react/components/Tabs/TabsContext";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCalendar } from "@fortawesome/free-solid-svg-icons";
+
+type AddFileProps = {
+  showAddFilePopUp: boolean;
+  setShowAddFilePopUp: Dispatch<SetStateAction<boolean>>;
+  seniorId: string;
+  files: PrismaFile[];
+  folder: string;
+};
+
+const TagOptions = ({
+  selectedTags,
+  setSelectedTags,
+}: {
+  selectedTags: TagProps[];
+  setSelectedTags: React.Dispatch<React.SetStateAction<TagProps[]>>;
+}) => {
+  return (
+    <div className="flex grid-flow-row flex-wrap">
+      {tagList.map((tag) => (
+        <div key={tag.name}>
+          <input
+            type="checkbox"
+            id={tag.name}
+            className="size-40 peer my-5 hidden"
+            onClick={() => {
+              if (!selectedTags.includes(tag) && selectedTags.length < 3) {
+                setSelectedTags([...selectedTags, tag]);
+              } else {
+                setSelectedTags(selectedTags.filter((i) => i != tag));
+              }
+            }}
+          />
+          {!selectedTags.includes(tag) && (
```

Comment:
Nit: We can merge the div for `!selectedTags` and `selectedTags`, since they have almost the similar styling. The conditional can be applied in className, i.e 

```tsx
className={`similar styling... ${selectedTags.includes(tags) ? "some styling" : "other styling"}`}
```